File: compare-pairs-and-display-differences/diff-pairs.rb
#!/usr/bin/env ruby

`ls #{ARGV[0]}| xargs -n 2`
  .split("\n")
  .each{|line|
    system "zcmp -s #{line}" or (
      puts "#{line} differ"
      system "vim -d #{line}"
    )
  }

File: compare-pairs-and-display-differences/diff-pairs.pl
#!/usr/bin/perl

my @a = split /\n/, `ls $ARGV[0] | xargs -n 2`;
foreach my $l (@a) {
	system('zcmp -s ' .  $l);
	if ($? != 0) {
		print "$l differ\n";
		system('vim -d ' . $l);
	}
}
